src/mlocale.cpp

```diff
@@ -4770,8 +4803,9 @@ QString MLocalePrivate::formatPhoneNumber( const QString& phoneNumber,
     MLocale::PhoneNumberGrouping grouping ) const
 {
   // first do sanity check of the input string
-  QRegExp rx( "\\+?\\d*" );
-  if ( ! rx.exactMatch( phoneNumber ) )
+  QRegularExpression rx( QRegularExpression::anchoredPattern("\\+?\\d*") );
```

Comment:
anchoredPattern() introduced in qt 5.15. We still need support for older versions too.